pkg/resource/db_instance/hooks.go

```diff
@@ -125,6 +125,16 @@ func instanceHasTerminalStatus(r *resource) bool {
 	return false
 }
 
+// instanceAvailable returns true if the supplied DB instance is in an
+// available status
+func instanceAvailable(r *resource) bool {
+	if r.ko.Status.DBInstanceStatus == nil {
+		return false
+	}
+	dbis := *r.ko.Status.DBInstanceStatus
+	return dbis == StatusAvailable
```

Comment:
Would this keep requeueing if the status is not "StatusAvailable" but something else like "StatusTerminal"?
In which case we would wanna stop requeueing.

Reply:
Well, this particular piece of code doesn't requeue or anything.. it's just for determining if the DB instance is available or not. There are hooks for sdk_update_pre_build_request that will set the CR to a Terminal state in some cases, though:

https://github.com/aws-controllers-k8s/rds-controller/blob/main/templates/hooks/db_instance/sdk_update_pre_build_request.go.tpl